example/lib/main.dart

```diff
@@ -447,44 +448,101 @@ class _MyAppState extends State<MyApp> {
-        onPressed: () async {
-          print('Sending rest message');
-          try {
-            await _rest.channels
-                .get('test')
-                .publish(name: 'Hello', data: 'Flutter $msgCounter');
-            print('Rest message sent.');
-            setState(() {
-              ++msgCounter;
-            });
-          } on ably.AblyException catch (e) {
-            print('Rest message sending failed:: $e :: ${e.errorInfo}');
-          }
-        },
+        onPressed: (_rest == null)
+            ? null
+            : () async {
+                print('Sending rest message');
+                try {
+                  await _rest.channels
+                      .get('test')
+                      .publish(name: 'Hello', data: 'Flutter $msgCounter');
+                  print('Rest message sent.');
+                  setState(() {
+                    ++msgCounter;
+                  });
+                } on ably.AblyException catch (e) {
+                  print('Rest message sending failed:: $e :: ${e.errorInfo}');
+                }
+              },
         color: Colors.yellow,
         child: const Text('Publish'),
       );
 
   Widget getRestChannelHistory() => FlatButton(
-        onPressed: () async {
-          final next = _restHistory?.hasNext() ?? false;
-      print('Rest history: getting ${next ? 'next' : 'first'} page');
-      try {
-        if (_restHistory == null || _restHistory.items.isEmpty) {
-          final result = await _rest.channels.get('test').history(
-            ably.RestHistoryParams(direction: 'forwards', limit: 10),
-          );
-          _restHistory = result;
-        } else if (next) {
-          _restHistory = await _restHistory.next();
-        } else {
-          _restHistory = await _restHistory.first();
-        }
-        setState(() {});
-      } on ably.AblyException catch (e) {
-            print('failed to get history:: $e :: ${e.errorInfo}');
-          }
-        },
+        onPressed: (_rest == null)
+            ? null
+            : () async {
+                final next = _restHistory?.hasNext() ?? false;
+                print('Rest history: getting ${next ? 'next' : 'first'} page');
+                try {
+                  if (_restHistory == null || _restHistory.items.isEmpty) {
+                    final result = await _rest.channels.get('test').history(
+                        ably.RestHistoryParams(
+                            direction: 'forwards', limit: 10));
+                    _restHistory = result;
+                  } else if (next) {
+                    _restHistory = await _restHistory.next();
+                  } else {
+                    _restHistory = await _restHistory.first();
+                  }
+                  setState(() {});
+                } on ably.AblyException catch (e) {
+                  print('failed to get history:: $e :: ${e.errorInfo}');
+                }
+              },
+        onLongPress: (_rest == null)
```

Comment:
Interesting. Is this behaviour obvious? Maybe I don't fully understand the context (I've not loaded the app in a while, least not for this branch), but I would usually expect a long-press to first present a menu or something.

Reply:
@QuintinWillison long press is not intuitive, I had a plan to enhance this, but primary focus was not on example.